src/toil/provisioners/abstractProvisioner.py

```diff
@@ -497,30 +501,37 @@ def addSSHRSAKey(self, keyData: str):
 
             self.sshPublicKeys.append("ssh-rsa " + keyData)
 
-
-        def toCloudConfig(self) -> str:
+        def toIgnitionConfig(self) -> str:
             """
-            Return a CloudConfig configuration describing the desired config.
+            Return an Ignition configuration describing the desired config.
             """
 
             # Define the base config
             config = {
-                'write_files': self.files,
-                'coreos': {
-                    'update': {
-                        'reboot-strategy': 'off'
-                    },
+                'ignition': {
+                    'version': '2.2.0'
+                },
+                'storage': {
+                    'files': self.files
+                },
+                'systemd': {
                     'units': self.units
                 }
             }
 
             if len(self.sshPublicKeys) > 0:
                 # Add SSH keys if needed
-                config['ssh_authorized_keys'] = self.sshPublicKeys
-
-            # Mark as CloudConfig and serialize as YAML
-            return '#cloud-config\n\n' + yaml.dump(config)
+                config['passwd'] = {
+                    'users': [
+                        {
+                            'name': 'core',
+                            'sshAuthorizedKeys': self.sshPublicKeys
+                        }
+                    ]
+                }
 
+            # Serialize as JSON
+            return json.dumps(config, separators=(',', ':'))
```

Comment:
We're getting pretty close to AWS's [16 KB limit](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-add-user-data.html) - We're writing the most amount of config when `clusterType == 'kubernetes' and role == 'leader'`, and we had ~14276 bytes previously with YAML. We are ~16244 bytes now (I think it depends on the cluster name too).